### DISCUSSION TITLE
Combating Early Bird and Increasing Auto Check In Efficiency

### DISCUSSION THREAD

**Discussion by Bubba8291** [jdholtz/auto-southwest-check-in#190] [Q&A] [answered]
I have been using [autoluv](https://github.com/byalextran/southwest-checkin) for Southwest auto check-ins. The first two flights I used with it, I got A31 and A38 both were full flights. This was back in February 2022. The flights after those two, I've been getting very high A's and low B's. I have a flight check-in tomorrow, and I will be using this program instead of the ruby program I referenced above to see if I notice a difference.

I am not sure if it is performance related since I think there might be delay with the ruby program. I have also noticed that flights I have gotten Early Bird with, I have gotten mid to high A's. As known, Early Bird check-ins happen automatically 36 hours in advance. We are not sure how the order of Early Bird check-ins are decided, but my thought is that it is either random or the order of the individual purchases of flights determines the Early Bird order.

That being said, Early Bird seems to have become more popular in past year or two. Since we cannot combat Early Bird, are there ways we can make the auto check-in more efficient? Maybe potentially reverse engineering the API so we can initiate a check-in 36 hours in advance if we have purchased Early Bird?

Something I did to add efficiency is that I set my time server to time.nist.gov (NIST operates the U.S. atomic clock). I am not sure if that will help or not though. Maybe figuring out the time server that Southwest uses might potentially help as well.

What are y'alls thoughts?

**Comment by jdholtz** [comment, marked as answer]
> are there ways we can make the auto check-in more efficient?

Currently, the program starts to check in 5 seconds before the check-in time in case your clock time is slightly different than Southwest's servers. That being said, I have never personally experienced it being off. My check-ins consistently go through the final request 8 seconds after check-in time and the only optimization I really see is to either reduce the time in between each failed request (currently 0.5 seconds, but I don't think reducing it will really help much) or speeding up your network connection.

> Maybe potentially reverse engineering the API so we can initiate a check-in 36 hours in advance if we have purchased Early Bird?

Could you explain more what you mean here? Early bird check-ins are done automatically, as you said above.

> I am not sure if that will help or not though.

As I mentioned above, the check-in starts 5 seconds early so I don't think that will help much.

There is a little more discussion/data on speeding up check-ins in #175.
